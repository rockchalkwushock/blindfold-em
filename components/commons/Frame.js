/* eslint-disable no-nested-ternary */
import Button from './Button'
import { Fragment } from '../../lib'
import Input from './Input'
import SubTitle from './SubTitle'

const Frame = ({ fn, name, next, prev, text, value }) => (
  <div>
    <SubTitle text={text} />
    <Input name={name} fn={fn} value={value} />
    {next && !prev ? (
      <Button className="next" fn={next} text="Next" />
    ) : (
      <Fragment>
        <Button className="next" fn={next} text="Next" />
        <Button className="prev" fn={prev} text="Prev" />
      </Fragment>
    )}
  </div>
)
export default Frame
