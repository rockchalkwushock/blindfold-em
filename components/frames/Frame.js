import PropTypes from 'prop-types'

import { Button, FlexContainer, Input, Span, SubTitle } from '../commons'

/**
 * @function Frame
 * @description renders single frame for Wizard Form.
 *
 * @prop {String} error
 * @prop {Function} fn
 * @prop {String} msg
 * @prop {String} name
 * @prop {Function} next
 * @prop {String} text
 * @prop {String} value
 * @returns React Element
 */
const Frame = ({ error, fn, msg, name, next, text, value }) => (
  <FlexContainer className="frame">
    <SubTitle text={text} />
    <Input fn={fn} name={name} msg={msg} value={value} />
    {error ? <Span msg={error} /> : null}
    <Button className="next" fn={next} text="Next" />
  </FlexContainer>
)

Frame.propTypes = {
  error: PropTypes.string,
  fn: PropTypes.func.isRequired,
  msg: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  next: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired
}

export default Frame
