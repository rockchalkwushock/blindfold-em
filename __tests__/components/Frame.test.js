import React from 'react'
import { mount } from 'enzyme'

import Frame from '../../components/commons/Frame'

describe('Component: <Frame />', () => {
  describe('Snapshots', () => {
    test('should mount without exploding as Frame 1', () => {
      const tree = mount(
        <Frame
          fn={() => {}}
          name="activity"
          next={() => {}}
          text="What activity are you working on?"
          value=""
        />
      )
      expect(tree).toMatchSnapshot()
    })
    test('should mount without exploding as Frame 2', () => {
      const tree = mount(
        <Frame
          fn={() => {}}
          name="timer"
          next={() => {}}
          prev={() => {}}
          text="How long should the timer be?"
          value=""
        />
      )
      expect(tree).toMatchSnapshot()
    })
    test('should mount without exploding as Frame 3', () => {
      const tree = mount(
        <Frame
          fn={() => {}}
          name="cooldown"
          next={() => {}}
          prev={() => {}}
          text="How long should the cooldown be?"
          value=""
        />
      )
      expect(tree).toMatchSnapshot()
    })
  })
})
