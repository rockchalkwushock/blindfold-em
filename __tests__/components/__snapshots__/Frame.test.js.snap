// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Frame /> Snapshots should mount without exploding as Frame 1 1`] = `
<Frame
  fn={[Function]}
  name="activity"
  next={[Function]}
  text="What activity are you working on?"
  value=""
>
  <div>
    <SubTitle
      text="What activity are you working on?"
    >
      <h2>
        What activity are you working on?
      </h2>
    </SubTitle>
    <Input
      fn={[Function]}
      name="activity"
      value=""
    >
      <div>
        <input
          name="activity"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </Input>
    <Button
      className="next"
      fn={[Function]}
      text="Next"
    >
      <button
        className="next"
        onClick={[Function]}
      >
        Next
      </button>
    </Button>
  </div>
</Frame>
`;

exports[`Component: <Frame /> Snapshots should mount without exploding as Frame 2 1`] = `
<Frame
  fn={[Function]}
  name="timer"
  next={[Function]}
  prev={[Function]}
  text="How long should the timer be?"
  value=""
>
  <div>
    <SubTitle
      text="How long should the timer be?"
    >
      <h2>
        How long should the timer be?
      </h2>
    </SubTitle>
    <Input
      fn={[Function]}
      name="timer"
      value=""
    >
      <div>
        <input
          name="timer"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </Input>
    <Fragment>
      <Button
        className="next"
        fn={[Function]}
        text="Next"
      >
        <button
          className="next"
          onClick={[Function]}
        >
          Next
        </button>
      </Button>
      <Button
        className="prev"
        fn={[Function]}
        text="Prev"
      >
        <button
          className="prev"
          onClick={[Function]}
        >
          Prev
        </button>
      </Button>
    </Fragment>
  </div>
</Frame>
`;

exports[`Component: <Frame /> Snapshots should mount without exploding as Frame 3 1`] = `
<Frame
  fn={[Function]}
  name="cooldown"
  next={[Function]}
  prev={[Function]}
  text="How long should the cooldown be?"
  value=""
>
  <div>
    <SubTitle
      text="How long should the cooldown be?"
    >
      <h2>
        How long should the cooldown be?
      </h2>
    </SubTitle>
    <Input
      fn={[Function]}
      name="cooldown"
      value=""
    >
      <div>
        <input
          name="cooldown"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </Input>
    <Fragment>
      <Button
        className="next"
        fn={[Function]}
        text="Next"
      >
        <button
          className="next"
          onClick={[Function]}
        >
          Next
        </button>
      </Button>
      <Button
        className="prev"
        fn={[Function]}
        text="Prev"
      >
        <button
          className="prev"
          onClick={[Function]}
        >
          Prev
        </button>
      </Button>
    </Fragment>
  </div>
</Frame>
`;
